lib/vm/src/vmcontext.rs

```diff
@@ -15,15 +15,37 @@ use std::ptr::{self, NonNull};
 use std::sync::Arc;
 use std::u32;
 
+/// We stop lying about what this daat is
+/// TODO:
+#[derive(Copy, Clone)]
+pub union VMFunctionExtraData {
+    /// Wasm function, it has a real VMContext:
+    pub vmctx: *mut VMContext,
+    /// Host functions can have custom environments
+    pub host_env: *mut std::ffi::c_void,
+}
+
+impl std::fmt::Debug for VMFunctionExtraData {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "FunctionExtarData union")
+    }
+}
```

Comment:
Do we need a custom Debug impl? At most we could see whether the pointer is null?

Reply:
You can't derive `Debug` on unions in Rust, but yeah we can definitely at least print some info here, and because it's really a union of exclusively pointers, we can honestly print the hex of the pointer too